packages/react-ui/src/app/builder/builder-page.tsx

```diff
@@ -12,14 +14,35 @@ import {
   ResizablePanelGroup,
 } from '@/components/ui/resizable-panel';
 import { RunDetailsBar } from '@/features/flow-runs/components/run-details-bar';
-
+import { cn } from '../../lib/utils';
 import { BuilderNavBar } from './builder-nav-bar';
 import { FlowVersionsList } from './flow-versions/flow-versions-list';
 import { PiecesCardList } from './pieces-list/pieces-card-list';
 import { FlowRunDetails } from './run-details/flow-run-details-list';
 import { FlowRecentRunsList } from './run-list/flow-runs-list';
 import { StepSettings } from './step-settings/step-settings-container';
 
+const minWidthOfSidebar = 'min-w-[max(20vw,400px)]';
+const useAnimateSidebar = (
+  isSidebarClosed: () => boolean,
+  sidebarValue: LeftSideBarType | RightSideBarType,
+) => {
+  const handleRef = useRef<ImperativePanelHandle>(null);
+  useEffect(() => {
+    if (
+      !isSidebarClosed() &&
+      (handleRef.current?.getSize() === 0 ||
+        handleRef.current?.getSize() === undefined)
+    ) {
+      handleRef.current?.resize(25);
+    } else if (isSidebarClosed()) {
+      console.log(handleRef.current);
+      handleRef.current?.resize(0);
+    }
+  }, [handleRef, isSidebarClosed, sidebarValue]);
+  return handleRef;
+};
```

Comment:
```suggestion
const useAnimateSidebar = (
  sidebarValue: LeftSideBarType | RightSideBarType,
) => {
  const handleRef = useRef<ImperativePanelHandle>(null);
  const sidebarbarClosed = [LeftSideBarType.NONE, RightSideBarType.NONE].includes(sidebarValue);
  useEffect(() => {
    const sidebarSize = handleRef.current?.getSize() ?? 0;
    if (sidebarbarClosed) {
      handleRef.current?.resize(0);
    } else if (sidebarSize === 0){
      handleRef.current?.resize(25);
    }
  }, [handleRef, sidebarValue]);
  return handleRef;
};
```
- I removed isSidebarClosed() and replaced it with boolean since you already have value.
- I moved the sidebarClosed check to the first condition so we don't have to check it in else.
- I removed isSidebarClosed() from the depdendencies array since it's not used.
- I extracted the sideBar and varriable and used it.